pkg/config/config.go

```diff
@@ -1,26 +1,75 @@
 package config
 
+import (
+	"fmt"
+
+	"github.com/caarlos0/env"
+	"github.com/golang/glog"
+)
+
 type DaemonConfig struct {
-	Plugin         string `json:"plugin"`          // Subnet manager plugin name
-	PeriodicUpdate int    `json:"periodic_update"` // Interval between every check for the added and deleted pods
-	GuidPool       GuidPoolConfig
-	SubnetManager  SubnetManagerPluginConfig
+	Plugin         string                    `json:"plugin" env:"PLUGIN"`                   // Subnet manager plugin name
+	PeriodicUpdate int                       `json:"periodic_update" env:"PERIODIC_UPDATE"` // Interval between every check for the added and deleted pods
+	GuidPool       GuidPoolConfig            `json:"guid_pool"`
+	SubnetManager  SubnetManagerPluginConfig `json:"subnet_manager"`
 }
 
 type GuidPoolConfig struct {
-	GuidRangeStart string `json:"guid_range_start"` // First guid of the pool
-	GuidRangeEnd   string `json:"guid_range_end"`   // Last of the guid pool
+	GuidRangeStart string `json:"guid_range_start" env:"GUID_RANGE_START"` // First guid of the pool
+	GuidRangeEnd   string `json:"guid_range_end" env:"GUID_RANGE_END"`     // Last of the guid pool
 }
 
 type SubnetManagerPluginConfig struct {
 	UFMConfig
```

Comment:
for this to be extendable its better to:

```
type SubnetManagerConfig struct {
    Name string 'json:"name"'
    Config json.RawMessage  'json:"config"'
}
```

or

```
type SubnetManagerConfig struct {
    ufm *UFMConfig 'json:"ufm",ommitempty'
}

```

anyway adding it flat is not good imo

Reply:
@adrianchiris Please check[here](https://github.com/Mellanox/ib-kubernetes/pull/3#discussion_r392674381)

Reply:
another suggestion as discussed with @moshe010 is to have all subnet manager configs in the same struct so something like:

```
type SubnetManagerConfig struct {
    PluginName string `env:plugin`
    ufm UFMConfig
}
```

Reply:
changed